### PR TITLE
[Merged by Bors] - feat: port FieldTheory.PolynomialGaloisGroup

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

- [x] depends on: #5341 

TODO:

- [x] timeouts
- [x] failed to synthesize instance `IsScalarTower ℚ ℝ ℂ`

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
I am going to work in this PR for a while

**Comment by xroblot** [comment]
I was able to fix a timeout (well, still needs 300000 heartbeats) but was unsuccessful with the `IsScalarTower` issue. I made a Zulip [thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/!4.235159.20FieldTheory.2EPolynomialGaloisGroup/near/368162465) to get some help.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5341
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5341~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/FieldTheory/PolynomialGaloisGroup.lean [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+- `Polynomial.Gal.card_complex_roots_eq_card_real_add_card_not_gal_inv`: The number of complex roots
+equals the number of real roots plus the number of roots not fixed by complex conjugation
+(i.e. with some imaginary component).
+
+-/
+
+
+example :=
+  AlgEquiv.restrictScalars ℚ Complex.conjAe
+
```
Comment (ChrisHughes24) [inline Mathlib/FieldTheory/PolynomialGaloisGroup.lean:49]:
```suggestion
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**